controller/eks-cluster-config-handler.go

```diff
@@ -1185,14 +1189,25 @@ func (h *Handler) updateUpstreamClusterState(upstreamSpec *eksv1.EKSClusterConfi
 			ClusterName:   aws.String(config.Spec.DisplayName),
 		}
 
+		// there exists a launch template for this node group
 		if upstreamNg.LaunchTemplate != nil {
 			upstreamTemplateVersion := aws.Int64Value(upstreamNg.LaunchTemplate.Version)
 			var err error
 			lt := ng.LaunchTemplate
 
+			// once the cluster is created, update the node group launch template with the default cluster security
+			// group generated by EKS
+			addDefaultClusterSecurityGroup := false
+			if !reflect.DeepEqual(config.Status.SecurityGroups, upstreamSpec.SecurityGroups) {
+				logrus.Infof("adding default security group to node group [%s] for cluster [%s]", aws.StringValue(upstreamNg.NodegroupName), config.Name)
+				addDefaultClusterSecurityGroup = true
+				config = config.DeepCopy()
+				config.Status.SecurityGroups = upstreamSpec.SecurityGroups
```

Comment:
WHy does this need to be done for every upstream nodegroup?

Reply:
Because every node instance in all the node groups for the cluster need to have the AWS cluster default security group attached, since it's cluster level

Reply:
but upstreamSpec isn't being updated so isn't one write of this enough? Like how would `!reflect.DeepEqual(config.Status.SecurityGroups, upstreamSpec.SecurityGroups)` ever be met after the first iteration?